play_book_pdf_tool/play_book_pdf_tool.py

```diff
@@ -90,14 +90,12 @@ def add_metadata(base_path, pdf):
     title, authors, pub_date, num_pages, publisher = itemgetter('title', 'authors', 'pub_date', 'num_pages',
                                                                 'publisher')(manifest['metadata'])
 
-    year, month, day = [int(x) for x in pub_date.split('.')]
-
     with pdf.open_metadata() as pdf_metadata:
         pdf_metadata['dc:title'] = title
         pdf_metadata['dc:creator'] = [author.strip() for author in authors.split(',')]
         pdf_metadata['dc:language'] = [language]
         pdf_metadata['dc:publisher'] = [publisher]
-        pdf_metadata['xmp:CreateDate'] = datetime.datetime(year, month, day).isoformat()
+        pdf_metadata['xmp:CreateDate'] = pub_date.replace('.', '-')
```

Comment:
```suggestion
        iso_8601_pub_date = pub_date.replace('.', '-')
        if re.match(r'^\d{4}-\d{2}-\d{2}$', iso_8601_pub_date):
                pdf_metadata['xmp:CreateDate'] = iso_8601_pub_date
        else:
                logging.warning(f"Invalid xmp:CreateDate format '{iso_8601_pub_date}'. Metadata will not include 'xmp:CreateDate'.")
```

Reply:
r'^\d{4}-\d{2}-\d{2}$' This does not solve the initial problem, when we can only have a year and a month in the date, without a day, or just a only year...

Reply:
@coderabbit Validate that the date is in the following format:

```
YYYY
YYYY-MM
YYYY-MM-DD
YYYY-MM-DDThh:mmTZD
YYYY-MM-DDThh:mm:ssTZD
YYYY-MM-DDThh:mm:ss.sTZD
```